netman/adapters/switches/brocade.py

```diff
@@ -148,16 +149,19 @@ def set_access_vlan(self, interface_id, vlan):
             if result:
                 raise UnknownInterface(interface_id)
 
-    def configure_native_vlan(self, interface_id, vlan):
+    def set_native_vlan(self, interface_id, vlan):
         return self.set_access_vlan(interface_id, vlan)
 
-    def openup_interface(self, interface_id):
-        with self.config(), self.interface(interface_id):
-            self.shell.do("enable")
+    def set_interface_state(self, interface_id, state):
+        if state is OFF:
+            with self.config(), self.interface(interface_id):
+                self.shell.do("disable")
+        elif state is ON:
+            with self.config(), self.interface(interface_id):
+                self.shell.do("enable")
+        else:
+            raise UnknownState(state)
```

Comment:
1 - This is not tested
2 - This might not be required...
state is here assumed to be part of an ENUM, therefore a user should use ON or OFF and nothing else
the api layer should validate ON or OFF has been passed and "cast" then into the "enum" so i don't think the method should validate this parameter

Thoughts?